packages/react-hooks/src/ConnectionProvider.tsx

```diff
@@ -15,49 +15,58 @@ import {
   addRecord,
 } from './recordUtils'
 
+export type useConnectionsOptions = {
+  excludedTypes?: [ConnectionType | string]
+  connectionState?: DidExchangeState
+}
+
 const ConnectionContext = createContext<RecordsState<ConnectionRecord> | undefined>(undefined)
 
-export const useConnections = () => {
+/**
+ * This method retreives the connection context for the current agent.
+ * From this you can access all connection records for the agent.
+ * @param options options for useConnections hook, lets us filter out specific types and limit states
+ * @returns a connection context containing information about the agents connections
+ */
+export const useConnections = (options: useConnectionsOptions = {}) => {
   const connectionContext = useContext(ConnectionContext)
   if (!connectionContext) {
     throw new Error('useConnections must be used within a ConnectionContextProvider')
   }
-  return connectionContext
-}
-
-export const useConnectionById = (id: string): ConnectionRecord | undefined => {
-  const { records: connections } = useConnections()
-  return connections.find((c: ConnectionRecord) => c.id === id)
-}
 
-export const useConnectionByState = (state: DidExchangeState | DidExchangeState[]): ConnectionRecord[] => {
-  const states = useMemo(() => (typeof state === 'string' ? [state] : state), [state])
-
-  const { records: connections } = useConnections()
+  let connections = connectionContext.records
+
+  // Only run this code if we don't have options.excludedTypes or options.connectionState to limit amount of loops
+  if (!options.connectionState && !options.excludedTypes) return { ...connectionContext, records: connections }
+
+  connections = useMemo(() => {
+    // do not filter if not filter options are provided to save on a loop
+    if (!options.connectionState && !options.excludedTypes) return connections
+
+      return connections.filter((record: ConnectionRecord) => {
+    // By default we include this connection
+    let valid = true
+    
+    
+    // Filter by state (if connectionState is defined)
+    if (options.connectionState) valid = record.state === options.connectionState
```

Comment:
Should this not just immediately return false if it doesn't match? If it doesn't match the state and it sets it to false _and_ then record types aren't excluded, valid will then get set to true, right?

Reply:
You're right, good catch 😄

Another option would be to do `valid = valid && <check>`